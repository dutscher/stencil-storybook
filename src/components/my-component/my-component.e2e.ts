import { newE2EPage } from '@stencil/core/testing';
import { getComponentNameFromFilename, setupE2E } from '../../utils/tests/setup';

const componentName = getComponentNameFromFilename(__filename);

describe(componentName, () => {
  it('default', async () => {
    const { page, component } = await setupE2E(componentName, null);
    await page.waitForChanges();
    expect(component).toMatchSnapshot();
  });

  it('default manually', async () => {
    const page = await newE2EPage();

    await page.setContent('<my-component></my-component>');
    const element = await page.find(componentName);
    expect(element).toHaveClass('hydrated');
  });

  it('renders changes to the name data', async () => {
    const page = await newE2EPage();

    await page.setContent('<my-component></my-component>');
    const component = await page.find(componentName);
    const element = await page.find(componentName + ' >>> div');
    expect(element.textContent).toEqual(`Hello, World! I'm `);

    component.setProperty('first', 'James');
    await page.waitForChanges();
    expect(element.textContent).toEqual(`Hello, World! I'm James`);

    component.setProperty('last', 'Quincy');
    await page.waitForChanges();
    expect(element.textContent).toEqual(`Hello, World! I'm James Quincy`);

    component.setProperty('middle', 'Earl');
    await page.waitForChanges();
    expect(element.textContent).toEqual(`Hello, World! I'm James Earl Quincy`);
  });
});
