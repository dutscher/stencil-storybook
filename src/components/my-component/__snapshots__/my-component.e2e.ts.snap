// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`my-component default 1`] = `
<my-component class="container dasdsa---dddsd hydrated my-component">
  <!---->
  Hello, World! I'm
  <span class="my-component__name"></span>
</my-component>
`;
